src/main/java/rx/internal/operators/OperatorTimeout.java

```diff
@@ -1,58 +0,0 @@
-/**
- * Copyright 2014 Netflix, Inc.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- * http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package rx.internal.operators;
-
-import java.util.concurrent.TimeUnit;
-
-import rx.*;
-import rx.functions.Action0;
-
-/**
- * Applies a timeout policy for each element in the observable sequence, using
- * the specified scheduler to run timeout timers. If the next element isn't
- * received within the specified timeout duration starting from its predecessor,
- * the other observable sequence is used to produce future messages from that
- * point on.
- * @param <T> the value type
- */
-public final class OperatorTimeout<T> extends OperatorTimeoutBase<T> {
```

Comment:
Is `OperatorTimeoutBase` still used somewhere?

Reply:
In `OperatorTimeoutWithSelector`.

Reply:
I probably should have mentioned in my original bug, but this variant of timeout (which uses `OperatorTimeoutWithSelector`) also exhibits the same problem:

```
public final <U, V> Observable<T> timeout(Func0<? extends Observable<U>> firstTimeoutSelector, Func1<? super T, ? extends Observable<V>> timeoutSelector, Observable<? extends T> other)
```

Can this fix be adapted to cover that variant as well (which we also use)?

Reply:
Yes, they were built on the same abstraction. It wasn't part of the initial report and I try to minimize my interaction with 1.x.

Reply:
Understood.  As I mention above, the new operator actually allows us to solve both of our use cases, so I'm fine with leaving the other operator alone.